plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -530,6 +528,52 @@ private static void handleStandardArgument(Props jobProps,
   }
 
   /**
+   * This method is used to get spark home from plugin's jobtype config.
+   * If spark.{sparkVersion}.home is set in commonprivate.properties/private.properties, then that will be returned.
+   * If spark.{sparkVersion}.home is not set and spark.home.dir is set then it will retrieve Spark directory inside
+   * spark.home.dir, matching spark.home.prefix + sparkVersion pattern. Spark directory name can be with dot
+   * in version name or without it. For e.g. spark-xxx-2_2105 or spark-xxx-2_2.1.0.5
+   * @param sparkVersion
+   * @return
+   */
+  private String getSparkHome(String sparkVersion) {
+    String sparkHome = getSysProps().get("spark." + sparkVersion + ".home");
+    if (sparkHome == null) {
+      info("Couldn't find spark." + sparkVersion + ".home property.");
+      String sparkDir = getSysProps().get("spark.home.dir");
+      String sparkHomePrefix =
+          getSysProps().get("spark.home.prefix") != null ? getSysProps().get("spark.home.prefix") : "*";
+      info(" Looking for spark at  " + sparkDir + " directory with " + sparkHomePrefix + " prefix for " + sparkVersion
+          + " version.");
+      DirectoryScanner scanner = new DirectoryScanner();
+      scanner.setBasedir(sparkDir);
+      scanner.setIncludes(
+          new String[]{sparkHomePrefix + sparkVersion.replace(".", "") + "*", sparkHomePrefix + sparkVersion + "*"});
+      scanner.scan();
+      String[] directories = scanner.getIncludedDirectories();
+      if (directories != null && directories.length > 0) {
+        sparkHome = sparkDir + "/" + directories[directories.length - 1];
+      } else {
+        scanner.setIncludes(new String[]{sparkHomePrefix + "*"});
+        scanner.scan();
+        String[] availableDirectories = scanner.getIncludedDirectories();
+        if (availableDirectories != null && availableDirectories.length > 0) {
+          Set<String> availableVersions = Sets.newTreeSet();
+          for (String directory : availableDirectories) {
+            String version = directory.replace(sparkHomePrefix, "");
+            availableVersions.add(version.contains(".") ? version.substring(0, 5)
+                : Joiner.on(".").join(version.substring(0, 3).split("")));
```

Comment:
This issue will occur even when we will want to keep both these binaries in Spark directory considering naming convention we are using.